dm/pkg/conn/utils.go

```diff
@@ -144,23 +141,63 @@ func GetMasterStatus(ctx *tcontext.Context, db *BaseDB, flavor string) (
 		| 0-1-2                    |
 		+--------------------------+
 	*/
+
+	var rowsResult [][]string
 	if flavor == gmysql.MySQLFlavor {
-		err = rows.Scan(&binlogName, &pos, &binlogDoDB, &binlogIgnoreDB, &gtidStr)
+		rowsResult, err = export.GetSpecifiedColumnValuesAndClose(rows, "File", "Position", "Binlog_Do_DB", "Binlog_Ignore_DB", "Executed_Gtid_Set")
+		if err != nil {
+			err = terror.DBErrorAdapt(err, terror.ErrDBDriverError)
+			return binlogName, pos, binlogDoDB, binlogIgnoreDB, gtidStr, err
+		}
+
+		if len(rowsResult) == 0 {
+			err = terror.ErrNoMasterStatus.Generate()
+			return binlogName, pos, binlogDoDB, binlogIgnoreDB, gtidStr, err
+		} else if len(rowsResult[0]) != 0 {
```

Comment:
will `len(rowsResult[0]) != 5` happen?

Reply:
According to the implementation of GetSpecifiedColumnValuesAndClose, `len(rowsResult[0])` is always equal to the number of the column names that are passed in, which is 5 here. It is possible that some of the retrieved values are "".

Reply:
then you can remove it or should check `!= 5` rather than `!= 0` and print a log when not so